[Bug]: Graceful rejection of token input for AWS Embeddings API

### What happened?

Hi Team, 

Love the project! First time contributing. 

I found a minor inconsistency in how input is handled across the OpenAI and AWS Embeddings API which results in a incorrect an APIConnectionError being thrown instead of a BadRequestError. 

In this report I provide some context on the API and the usage of it, provided a stack trace, provided code to reproduce the bug, and included my idea for a fix. With the maintainers approval I can implement my suggested fix.

# Background
The OpenAI Embeddings API accepts as input: str | List[str] | List[int] | List[List[int]]. 

However, the Bedrock Embeddings API for `amazon.titan-embed-text-v1` accepts only `str | List[str]`.

# Problem
Passing input=List[int] into `embedding(model='$OPENAI_MODEL')` is valid, but when passed into `embedding(model='amazon.titan-embed-text-v1')` results in:

```
> raise APIConnectionError(
      message=f"{str(original_exception)}",
      llm_provider=custom_llm_provider,
      model=model,
      request=httpx.Request(
          method="POST", url="https://api.openai.com/v1/"
      ),  # stub the request
  )
E litellm.exceptions.APIConnectionError: 'int' object has no attribute 'replace'
```
 
Passing input=List[List[int]] into `embedding(model='$OPENAI_MODEL')` is valid, but when passed into `embedding(model='amazon.titan-embed-text-v1')` results in:

```
> raise APIConnectionError(
      message=f"{str(original_exception)}",
      llm_provider=custom_llm_provider,
      model=model,
      request=httpx.Request(
          method="POST", url="https://api.openai.com/v1/"
      ),  # stub the request
  )
E litellm.exceptions.APIConnectionError: 'list' object has no attribute 'replace'
```

(Stack trace included in *relevant log output*).

Following the Stacktrace shows that base exception comes from input preprocessing at [llms/bedrock.py#L716](https://github.com/BerriAI/litellm/blob/main/litellm/llms/bedrock.py#L716). This line assumes that the input for Amazon LLM's is type str.

```
model = 'amazon.titan-embed-text-v1', input = [1]
client = <botocore.client.BedrockRuntime object at 0x1201f67a0>
optional_params = {}, encoding = None
logging_obj = <litellm.utils.Logging object at 0x117c0d450>

    def _embedding_func_single(
        model: str,
        input: str,
        client: Any,
        optional_params=None,
        encoding=None,
        logging_obj=None,
    ):
        # logic for parsing in - calling - parsing out model embedding calls
        ## FORMAT EMBEDDING INPUT ##
        provider = model.split(".")[0]
        inference_params = copy.deepcopy(optional_params)
        inference_params.pop(
            "user", None
        )  # make sure user is not passed in for bedrock call
        modelId = (
            optional_params.pop("model_id", None) or model
        )  # default to model if not passed
        if provider == "amazon":
>           input = input.replace(os.linesep, " ")
E           AttributeError: 'list' object has no attribute 'replace'
```

I think this should be a BadRequestError instead of a APIConnectionError.

# Suggested Resolution: Update Bedrock Embedding API to fail-fast on invalid input type

My proposed change would be add a line around [litellm/llms/bedrock.py#L810](https://github.com/BerriAI/litellm/blob/main/litellm/llms/bedrock.py#L810) to verify that it is iterating over List[str] and if the iterated member is not instanceof str then raise an BadRequestError with message 'Bedrock Embedding API input must be type str | List[str]'.

This change will work for the 3 models that LiteLLM has documented as supporting: `Titan Embeddings - G1`, `Cohere Embeddings - English`, and `Cohere Embeddings - Multilingual`. Based on the provided examples in the AWS Console all 3 of these models only support text input.

First thing to consider: Bedrock also a multimodal model named `Titan Multimodal Embeddings Generation 1`. Which has as input:
```
{
  "modelId": "amazon.titan-embed-image-v1",
  "contentType": "application/json",
  "accept": "application/json",
  "body": {
    "inputText": "this is where you place your input text",
    "inputImage": "<base64_image_string>"
  }
}
```
This API isn't supported by LiteLLM, and OpenAI doesn't offer multimodal embeddings, but if OpenAI does offer multimodal embeddings in the future then the above added typecheck would need to be removed.

I think this is acceptable risk as the type check can be updated to confirm the iterated item confirms to the new required type signature.

Second thing to consider: Could there be a wider type checking system in place to handle future API divergences? I think this is a good question but it's a decision which can come later as the above code does not create any one way doors.

**If you agree with this approach please confirm and I can implement the fix and send a PR**

# Reproducible unit test

Running `litellm-1.17.16`

```
import pytest
import litellm

def test_demo_tokens_as_input_to_embeddings_fails_for_titan():
    # Works as expected
    response_for_input_as_list_str = litellm.embedding(
        model='amazon.titan-embed-text-v1',
        input=['Hello world'],
        aws_region_name=AWS_REGION_NAME,
        aws_role_name=AWS_ROLE_NAME,
        aws_session_name=AWS_SESSION_NAME
    )

    assert len(response_for_input_as_list_str.data) == 1

    # Works as expected
    response_for_input_as_tokens_openai = litellm.embedding(
        model='text-embedding-ada-002',
        input=[1, 2, 3],
        api_key=OPENAI_TOKEN
    )

    assert len(response_for_input_as_tokens_openai.data) == 1

    # Works as expected
    response_for_input_as_list_of_tokens_openai = litellm.embedding(
        model='text-embedding-ada-002',
        input=[[1]],
        api_key=OPENAI_TOKEN
    )

    assert len(response_for_input_as_list_of_tokens_openai.data) == 1

    with pytest.raises(litellm.APIConnectionError, match="'list' object has no attribute 'replace'"):
        litellm.embedding(
            model='amazon.titan-embed-text-v1',
            input=[[1]],
            aws_region_name=AWS_REGION_NAME,
            aws_role_name=AWS_ROLE_NAME,
            aws_session_name=AWS_SESSION_NAME
        )

    with pytest.raises(litellm.APIConnectionError, match="'int' object has no attribute 'replace'"):
        litellm.embedding(
            model='amazon.titan-embed-text-v1',
            input=[1],
            aws_region_name=AWS_REGION_NAME,
            aws_role_name=AWS_ROLE_NAME,
            aws_session_name=AWS_SESSION_NAME
        )
```

### Relevant log output

_No response_

### Twitter / LinkedIn details

_No response_

--- a/litellm/llms/bedrock.py
+++ b/litellm/llms/bedrock.py
@@ -702,7 +702,7 @@ def _embedding_func_single(
     encoding=None,
     logging_obj=None,
 ):
-    if type(input) != str:
+    if isinstance(input, str) is False:
         raise BedrockError(
             message="Bedrock Embedding API input must be type str | List[str]",
             status_code=400,
@@ -800,7 +800,8 @@ def embedding(
         aws_role_name=aws_role_name,
         aws_session_name=aws_session_name,
     )
-    if type(input) == str:
+    if isinstance(input, str):
+        ## Embedding Call
         embeddings = [
             _embedding_func_single(
                 model,
@@ -810,8 +811,8 @@ def embedding(
                 logging_obj=logging_obj,
             )
         ]
-    elif type(input) == list:
-        ## Embedding Call
+    elif isinstance(input, list):
+        ## Embedding Call - assuming this is a List[str]
         embeddings = [
             _embedding_func_single(
                 model,

--- a/litellm/llms/bedrock.py
+++ b/litellm/llms/bedrock.py
@@ -702,6 +702,11 @@ def _embedding_func_single(
     encoding=None,
     logging_obj=None,
 ):
+    if type(input) != str:
+        raise BedrockError(
+            message="Bedrock Embedding API input must be type str | List[str]",
+            status_code=400,
+        )
     # logic for parsing in - calling - parsing out model embedding calls
     ## FORMAT EMBEDDING INPUT ##
     provider = model.split(".")[0]
@@ -805,7 +810,7 @@ def embedding(
                 logging_obj=logging_obj,
             )
         ]
-    else:
+    elif type(input) == list:
         ## Embedding Call
         embeddings = [
             _embedding_func_single(
@@ -817,6 +822,12 @@ def embedding(
             )
             for i in input
         ]  # [TODO]: make these parallel calls
+    else:
+        # enters this branch if input = int, ex. input=2
+        raise BedrockError(
+            message="Bedrock Embedding API input must be type str | List[str]",
+            status_code=400,
+        )
 
     ## Populate OpenAI compliant dictionary
     embedding_response = []

--- a/litellm/tests/test_embedding.py
+++ b/litellm/tests/test_embedding.py
@@ -302,6 +302,25 @@ def test_bedrock_embedding_cohere():
 # test_bedrock_embedding_cohere()
 
 
+def test_demo_tokens_as_input_to_embeddings_fails_for_titan():
+    litellm.set_verbose = True
+
+    with pytest.raises(
+        litellm.BadRequestError,
+        match="BedrockException - Bedrock Embedding API input must be type str | List[str]",
+    ):
+        litellm.embedding(model="amazon.titan-embed-text-v1", input=[[1]])
+
+    with pytest.raises(
+        litellm.BadRequestError,
+        match="BedrockException - Bedrock Embedding API input must be type str | List[str]",
+    ):
+        litellm.embedding(
+            model="amazon.titan-embed-text-v1",
+            input=[1],
+        )
+
+
 # comment out hf tests - since hf endpoints are unstable
 def test_hf_embedding():
     try:

